python/ray/serve/tests/test_deployment_in_container.py

```diff
@@ -0,0 +1,256 @@
+SERVE_CONTAINER_RUNTIME_ENV = {
+    "container": {
+        "image": "zcin/runtime-env-prototype:ci",
+        "worker_path": "/home/ray/anaconda3/lib/python3.8/site-packages/ray/_private/workers/default_worker.py",  # noqa
+    }
+}
+
+
+@pytest.mark.skip
+def test_basic(ray_start_stop):
+    @serve.deployment(ray_actor_options={"runtime_env": SERVE_CONTAINER_RUNTIME_ENV})
+    class Model:
+        def __call__(self):
+            with open("file.txt") as f:
+                return f.read()
+
+    def check_application(app_handle: DeploymentHandle, expected: str):
+        ref = app_handle.remote()
+        assert ref.result() == expected
+        return True
+
+    h = serve.run(Model.bind())
+    wait_for_condition(
+        check_application,
+        app_handle=h,
+        expected="Hi I'm Cindy, this is version 1\n",
+        timeout=300,
+    )
+
+
+@pytest.mark.skipif(sys.platform != "linux", reason="Only works on Linux.")
+def test_put_get(shutdown_only):
+    @ray.remote(runtime_env=PLAIN_CONTAINER_RUNTIME_ENV)
+    # @ray.remote
+    def create_ref():
+        ref = ray.put(np.zeros(100_000_000))
+        return ref
+
+    ray.init()
+    wrapped_ref = create_ref.remote()
+    ray.get(ray.get(wrapped_ref)) == np.zeros(100_000_000)
+
+
+@pytest.mark.skipif(sys.platform != "linux", reason="Only works on Linux.")
+def test_shared_memory(shutdown_only):
+    @ray.remote(runtime_env=PLAIN_CONTAINER_RUNTIME_ENV)
+    # @ray.remote
+    def f():
+        array = np.random.rand(5000, 5000)
+        return ray.put(array)
+
+    ray.init()
+    ref = ray.get(f.remote())
+    val = ray.get(ref)
+    assert sys.getsizeof(val) < 4 * 5000 * 5000
+    assert val.shape == (5000, 5000)
+
+
+@pytest.mark.skipif(sys.platform != "linux", reason="Only works on Linux.")
+def test_log_file_exists(shutdown_only):
+    """Verify worker log file exists"""
+    ray.init(num_cpus=1)
+
+    session_dir = ray._private.worker.global_worker.node.address_info["session_dir"]
+    session_path = Path(session_dir)
+    log_dir_path = session_path / "logs"
+
+    def task_finished():
+        tasks = list_tasks()
+        assert len(tasks) > 0
+        assert tasks[0].worker_id
+        assert tasks[0].worker_pid
+        assert tasks[0].state == "FINISHED"
+        return True
+
+    # Run a basic workload.
+    @ray.remote(runtime_env=PLAIN_CONTAINER_RUNTIME_ENV)
+    # @ray.remote
+    def f():
+        for i in range(10):
+            print(f"test {i}")
+
+    f.remote()
+    wait_for_condition(task_finished)
+
+    task_state = list_tasks()[0]
+    worker_id = task_state.worker_id
+    worker_pid = task_state.worker_pid
+    print(f"Worker ID: {worker_id}")
+    print(f"Worker PID: {worker_pid}")
+
+    paths = [path.name for path in log_dir_path.iterdir()]
+    assert f"python-core-worker-{worker_id}_{worker_pid}.log" in paths
+    assert any(re.search(f"^worker-{worker_id}-.*-{worker_pid}.err$", p) for p in paths)
```

Comment:
Can you help me understand what we're testing with this test?

Reply:
This checks that the log files exists for a worker that's started in its own container.